pkg/standalone-utils/test/VaultActions.test.ts

```diff
@@ -1161,4 +1190,324 @@ describe('VaultActions', function () {
+                    ops: [
+                      {
+                        kind: UserBalanceOpKind.TransferInternal,
+                        asset: tokens.DAI.address,
+                        amount: toChainedReference(0),
+                        sender: relayer.address,
+                        recipient: TypesConverter.toAddress(sender),
+                      },
+                      {
+                        kind: UserBalanceOpKind.TransferInternal,
+                        asset: tokens.SNX.address,
+                        amount: toChainedReference(1),
+                        sender: relayer.address,
+                        recipient: TypesConverter.toAddress(recipient),
+                      },
+                    ],
+                  }),
+                ]),
+              tokens,
+              [
+                {
+                  account: TypesConverter.toAddress(sender),
+                  changes: {
+                    DAI: amountDAI,
+                    SNX: 0,
+                  },
+                },
+                {
+                  account: TypesConverter.toAddress(recipient),
+                  changes: {
+                    DAI: 0,
+                    SNX: amountSNX,
+                  },
+                },
+              ],
+              vault.instance
+            )
+          ).wait();
+
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(sender),
+            token: tokens.DAI.address,
+            delta: amountDAI,
+          });
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(relayer),
+            token: tokens.DAI.address,
+            delta: amountDAI,
+          });
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(relayer),
+            token: tokens.SNX.address,
+            delta: amountSNX,
+          });
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(recipient),
+            token: tokens.SNX.address,
+            delta: amountSNX,
+          });
+        });
+
+        it('allows emergency exit', async () => {
+          const BPT = (await getBPT(poolIdA)).get(0).instance;
+          BPT.connect(user).transfer(TypesConverter.toAddress(sender), await BPT.balanceOf(user.address));
+
+          const amountInBPT = fp(1);
+
+          // Exit Pool A (DAI, MKR) to internal balance
+          // Pretend MKR is bricked (i.e., external transfers fail)
+          // Swap *internally* with Pool B MKR -> SNX
+          // Withdraw DAI and SNX back out to wallet
+          // So external token balances of DAI/MKR should be unchanged, and SNX should equal token out from swap
```

Comment:
Are we withdrawing DAI though?
I understand DAI is not withdrawn, so DAI/MKR external token balances are unchanged indeed.